pkg/service/aerospike_service.go

```diff
@@ -16,6 +18,75 @@ import (
 
 const namespaceInfo = "namespaces"
 
+type NamespaceValidator interface {
+	// MissingNamespaces returns a slice containing any namespaces specified in the
+	// provided slice which do not exist on the given cluster.
+	MissingNamespaces(cluster *model.AerospikeCluster, namespaces []string) []string
+	// ValidateRoutines verifies that all namespaces referenced in backup routines
+	// exist in their respective clusters.
+	ValidateRoutines(cluster *model.AerospikeCluster, config *model.Config) error
+}
+
+type defaultNamespaceValidator struct {
+	ClientManager ClientManager
+}
+
+func NewNamespaceValidator(clientManager ClientManager) NamespaceValidator {
+	return &defaultNamespaceValidator{
+		ClientManager: clientManager,
+	}
+}
+
+func (nv *defaultNamespaceValidator) MissingNamespaces(
+	cluster *model.AerospikeCluster,
+	namespaces []string,
+) []string {
+	if len(namespaces) == 0 {
+		return nil
+	}
+
+	backupClient, err := nv.ClientManager.GetClient(cluster)
+	if err != nil {
+		slog.Info("Failed to connect to aerospike cluster", slog.Any("error", err))
+		return nil
+	}
+	defer nv.ClientManager.Close(backupClient)
+
+	namespacesInCluster, err := getAllNamespacesOfCluster(backupClient.AerospikeClient())
+	if err != nil {
+		slog.Info("Failed to retrieve namespaces from cluster", slog.Any("error", err))
+	}
+
+	return util.MissingElements(namespaces, namespacesInCluster)
+}
+
+func (nv *defaultNamespaceValidator) ValidateRoutines(cluster *model.AerospikeCluster, config *model.Config) error {
+	var err error
+	routines := filterRoutinesByCluster(config.BackupRoutines, cluster)
+	for routineName, routine := range routines {
+		missingNamespaces := nv.MissingNamespaces(cluster, routine.Namespaces)
```

Comment:
This will create a new client each time, even though the routines are likely to share the cluster.

Reply:
no, we use nv.ClientManager.GetClient inside, so client will be taken from cache.

Reply:
It is being closed and removed from the cache right after the use.